### ISSUE TITLE
GDScript parser error "returning: __null" when triggering a specific error

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot#15906]
**Godot version:**
00387cb

**Issue description:**
When doing the following piece of code:
```
func test():
	# Insert '(' here
	var1 = 0
	if true:
		var2 = 0
		if true:
			pass
```
It will naturally alert that this is invalid, but will also print this error message:
`modules/gdscript/gdscript_parser.cpp:1303 - Method/Function Failed, returning: __null`
And I don't think this part is normal.

**Comment by poke1024** [comment]
Not a bug, part of the code:

```
				_set_error("Expected else after ternary if.");
				ERR_FAIL_V(NULL);
```

There are several other occasions where this happens. We could replace these `ERR_FAIL_V` with something else some time.

**Comment by vnen** [comment]
I solved this in #19264.

**Comment by vnen** [comment]
If needed, some patch can be applied to 3.0 to fix this in specific.
